### ISSUE TITLE
Mounting an ISO file (CD) or ejecting a floppy disk image results in a buffer overflow (SIGABRT) 

### ISSUE THREAD

**Issue by TealSwiftfox** [flathub/net._86box._86Box#18]
86Box 4.2.1 (Flatpak)
Fedora Silverblue 40 - GNOME 46
AMD Ryzen 5 4600G

86Box buffer overflows (and crashes) as soon as I mount any ISO file, same thing happens if I try to eject any IMG file.

Steps:

1. Open 86Box
2. Adding a 1.44 MB floppy drive and a CD-ROM drive, and configure them from the BIOS.
3. I add an ISO (or ejecting a mounted floppy disk image) and the log only says "*** buffer overflow detected ***: terminated"

Rolling back to the previous commit (86Box 4.2) the problem no longer occurs.

I add the coredump, just to add more information:

```
Stack trace of thread 2:
              #0  0x00007fde4eaa3e14 n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x90e14)
              #1  0x00007fde4ea51dce n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x3edce)
              #2  0x00007fde4ea3983f n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x2683f)
              #3  0x00007fde4ea3a710 n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x27710)
              #4  0x00007fde4eb380a9 n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x1250a9)
              #5  0x00007fde4eb37a64 n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x124a64)
              #6  0x00007fde4eb39267 n/a (/usr/lib/x86_64-linux-gnu/libc.so.6 + 0x126267)
              #7  0x000056485e199ad2 n/a (/app/bin/86Box + 0x88bad2)
              #8  0x000056485e19c167 n/a (/app/bin/86Box + 0x88e167)
              #9  0x000056485e1315b5 n/a (/app/bin/86Box + 0x8235b5)
              #10 0x000056485e131b35 n/a (/app/bin/86Box + 0x823b35)
              #11 0x000056485e131e24 n/a (/app/bin/86Box + 0x823e24)
              #12 0x00007fde4f5186f2 n/a (/usr/lib/x86_64-linux-gnu/libQt5Core.so.5.15.10 + 0x3186f2)
              #13 0x00007fde50174502 n/a (/usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5.15.10 + 0x174502)
              ELF object binary architecture: AMD x86-64
```

**Comment by AsciiWolf** [comment]
Thanks for your report! I was not able to reproduce this when mounting the .iso, however it did happen when trying to umount the mounted iso.

It looks like an upstream issue unrelated to this (unofficial) Flatpak. You can try reporting it [here](https://github.com/86Box/86Box/issues).

I can revert the Flatpak (and ROMs) to a previous version if this is a bigger issue preventing the normal use of 86Box.

**Comment by TealSwiftfox** [comment]
I compiled 86Box in a Distrobox container, and the application works correctly, without errors.
I tried the Flatpak version in a machine with a fresh install of Fedora, always a buffer overflow.

**Comment by AsciiWolf** [comment]
Is it possible to use Distrobox container with the freedesktop runtime as the base system? It would be worth trying.

**Comment by ssokolow** [comment]
Last week, the [upstream report](https://github.com/86Box/86Box/issues/4872) got "Also, this bug sounds familiar - I believe it was fixed in 4.2.2." as part of its reply.

Any chance we could get a version bump?

**Comment by AsciiWolf** [comment]
Yep, we can. After 4.2.2 gets [released](https://github.com/86Box/86Box/releases). :-)

**Comment by ssokolow** [comment]
\*facepalm*

Sorry for not double-checking that.

**Comment by KenjiUmino** [comment]
can confirm it also happens on steam deck with SteamOS running in desktop mode 

here is some log output if that is of any use:

`(deck@steamdeck ~)$ /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=86Box.sh net._86box._86Box
Qt: Session management error: Could not open network socket
#
# 86Box v4.2.1 logfile, created 2024/10/09 16:05:10
#
# VM: 86Box
#
# Emulator path: /app/bin/
# Userfiles path: /home/deck/.var/app/net._86box._86Box/config/86Box/
# ROM path: /home/deck/.var/app/net._86box._86Box/config/86Box/roms/
# ROM path: /app/bin/roms/
# ROM path: /home/deck/.var/app/net._86box._86Box/data/86Box/roms/
# ROM path: /app/share/86Box/roms/
# ROM path: /usr/share/86Box/roms/
# ROM path: /usr/share/runtime/share/86Box/roms/
# ROM path: /run/host/user-share/86Box/roms/
# ROM path: /run/host/share/86Box/roms/
# Configuration file: /home/deck/.var/app/net._86box._86Box/config/86Box/86box.cfg
#

Qt: version 5.15.10, platform "xcb"
Failed to load vulkan: Cannot load library vulkan: (vulkan: cannot open shared object file: No such file or directory)
initInstance: No Vulkan library available
Failed to create platform Vulkan instance
*** buffer overflow detected ***: terminated
(deck@steamdeck ~)$
`

it happens when unmounting a floppy or CD-ROM image or a folder mounted as CD-ROM, however, I am not sure if this is an upstream issue, if it was, it must have been fixed by upstream after the current 4.2.1 flatpak version was built, because the latest .appimage from their github page (namely v4.2.1 [build 6130]) does not have this issue on the same system ... neither does the older appimage v4.1.1 [build 5634] that I also tested.

**Comment by DekosAnjo** [comment]
Same here. I was trying to mount an ISO, and it crashes every time with the error message: "*** buffer overflow detected ***: terminated."

**Comment by OBattler** [comment]
Yes, there was indeed an overflow in 4.2.1, masked most of the time because there was valid non-protected memory after that in the heap, but the Flatpak resulted in a configuration where there was no such valid memory after that in the heap, revealing the problem. It has indeed been fixed at some point after the 4.2.1 release.

**Comment by AsciiWolf** [comment]
Do you remember what commit fixed this? I can add it as a patch.

**Comment by LuanVSO** [comment]
looks like it was 6b05602ea4d1593edf8d0fff17e733b182e94637

**Comment by AsciiWolf** [comment]
It looks like that [the commit](https://github.com/86Box/86Box/commit/6b05602ea4d1593edf8d0fff17e733b182e94637) sadly does not apply cleanly as a patch over the v4.2.1 86box version code. If you want, feel free to prepare a Pull request. I sadly do not have enough time I could allocate to work on this at the moment.

**Comment by AsciiWolf** [comment]
Should be fixed in 5.0.

**Comment by DekosAnjo** [comment]
yes, it has been fixed. now it works fine.
